src/main/groovy/com/sequenceiq/ambari/client/services/ClusterService.groovy

```diff
@@ -218,6 +218,31 @@ trait ClusterService extends CommonService {
     })
   }
 
+  /**
+   * Retire a host component on a given host.
+   * @param host hostName where the component is installed to
+   * @param serviceName the service to retire
+   * @param componentName the name of the component
+   */
+  def int retire(List<String> hosts, String serviceName, String componentName)
+          throws URISyntaxException, ClientProtocolException, HttpResponseException, IOException {
+    def requestInfo = [
+            command   : 'RETIRE',
+            context   : "Execute retire command.",
+            parameters: ['excluded_hosts': hosts.join(',')]
+    ]
+    def filter = [
+            ['service_name': serviceName, 'component_name': componentName]
```

Comment:
in the example, there is a third field, `hosts`
```
certs $ curl --insecure -u admin:admin1234 -X POST -H 'X-Requested-By:ambari' -H"Cookie: AMBARISESSIONID=node01wlt8eg7buea04wvjyq7yfs0812.node0" -d'{"RequestInfo":{"context":"Execute an action", "command" : "RETIRE"}, "Requests/resource_filters":[{"service_name": "NIFI", "component_name":"NIFI_MASTER","hosts":"ctr-e163-1618833847257-1055-01-000006.hwx.site"}] }' https://172.27.26.70/api/v1/clusters/mpatt2-1v78n8/requests
{
  "href" : "http://172.27.26.70/api/v1/clusters/mpatt2-1v78n8/requests/14",
  "Requests" : {
    "id" : 14,
    "status" : "Accepted"
  }
}
```

Reply:
yes, fixed